broker.go

```diff
@@ -0,0 +1,726 @@
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/husio/kafka/proto"
```

Comment:
I can close this pull request, but I don't understand what you want me to do instead.